monitoring/uss_qualifier/scd/executor/target.py

```diff
@@ -0,0 +1,51 @@
+
+from typing import Dict
+
+from monitoring.monitorlib import infrastructure, auth
+from monitoring.monitorlib.clients.scd_automated_testing import create_flight, delete_flight
+from monitoring.monitorlib.scd_automated_testing.scd_injection_api import InjectFlightResult, \
+    DeleteFlightResult, InjectFlightResponse, DeleteFlightResponse
+from monitoring.uss_qualifier.rid.utils import InjectionTargetConfiguration
+from monitoring.uss_qualifier.scd.data_interfaces import FlightInjectionAttempt
+
+
+class TestTarget():
```

Comment:
I'd recommend implementing the concept of a "dry run" almost entirely within this class.  One approach could be to have a `dry_run` flag that, when set, causes this class not to make any calls to the actual target, and instead always assume failure.  But actually, that's not quite what we want because often times, inject_flight should return a result other than Failed -- so maybe a `dry_run` flag isn't enough, and instead we could add something like `mocked_responses` which is a list of `Tuple[InjectFlightRequest, InjectFlightResponse]` and the appropriate response is looked up there according to the request received (instead of calling out to the real client).  But that seems to be getting pretty complicated -- it actually seems simpler to just spin up a few mock_uss instances and run the uss_qualifier normally.  Do we really need to be able to perform dry runs instead of running normally against mocks?

Reply:
I started to look into mocking the responses as you proposed. I will keep it for another PR if this is ok for you.

Reply:
Definitely -- I think smaller, separable PRs are best in general, subject to the fixed overhead of creating the PR and getting it reviewed